smarts/env/wrappers/standard_obs.py

```diff
@@ -0,0 +1,582 @@
+            "lane_index": gym.spaces.Box(low=0, high=127, shape=(_NEIGHBOR_SHP,), dtype=np.int8),
+            "pos": gym.spaces.Box(low=-1e10, high=1e10, shape=(_NEIGHBOR_SHP,3), dtype=np.float32),    
+            "speed": gym.spaces.Box(low=0, high=1e10, shape=(_NEIGHBOR_SHP,), dtype=np.float32),
+        }),
+        "ogm": lambda val: gym.spaces.Box(low=0, high=255,shape=(val.height, val.width, 1), dtype=np.uint8),
+        "rgb": lambda val: gym.spaces.Box(low=0, high=255, shape=(val.height, val.width, 3), dtype=np.uint8),
+        "ttc": lambda _: gym.spaces.Dict({
+            "angle_error": gym.spaces.Box(low=-np.pi, high=np.pi, shape=(1,), dtype=np.float32),
+            "distance_from_center": gym.spaces.Box(low=-1e10, high=1e10, shape=(1,), dtype=np.float32),
+            "dtc": gym.spaces.Box(low=-1e10, high=1e10, shape=(3,), dtype=np.float32),
+            "ttc": gym.spaces.Box(low=0, high=1e10, shape=(3,), dtype=np.float32),
+        }),
+        "waypoints": lambda _: gym.spaces.Dict({
+            "heading": gym.spaces.Box(low=-math.pi, high=math.pi, shape=_WAYPOINT_SHP, dtype=np.float32),
+            "lane_index": gym.spaces.Box(low=0, high=127, shape=_WAYPOINT_SHP, dtype=np.int8),
+            "lane_width": gym.spaces.Box(low=0, high=1e10, shape=_WAYPOINT_SHP, dtype=np.float32),
+            "pos": gym.spaces.Box(low=-1e10, high=1e10, shape=_WAYPOINT_SHP + (3,), dtype=np.float32),
+            "speed_limit": gym.spaces.Box(low=0, high=1e10, shape=_WAYPOINT_SHP, dtype=np.float32),
+        }),
+    }
+    # fmt: on
+
+    return basic, opt
+
+
+def _make_space(intrfcs: Dict[str, Any]) -> gym.Space:
+    space, opt_space = get_spaces()
+
+    for intrfc, val in intrfcs.items():
+        opt_ob = intrfc_to_stdobs(intrfc)
+        if opt_ob:
+            space.update({opt_ob: opt_space[opt_ob](val)})
+
+    if "waypoints" in intrfcs.keys() and "neighborhood_vehicles" in intrfcs.keys():
+        space.update({"ttc": opt_space["ttc"](None)})
+
+    return space
+
+
+def _std_dagm(
+    val: Optional[DrivableAreaGridMap],
+) -> Optional[np.ndarray]:
+    if not val:
+        return None
+    return val.data.astype(np.uint8)
+
+
+def _std_dist(val: float) -> float:
+    return np.float32(val)
+
+
+def _std_ego(
+    val: EgoVehicleObservation,
+) -> Dict[str, Union[np.float32, np.ndarray]]:
+    return {
+        "angular_acceleration": val.angular_acceleration.astype(np.float32),
+        "angular_jerk": val.angular_jerk.astype(np.float32),
+        "angular_velocity": val.angular_acceleration.astype(np.float32),
+        "box": np.array(val.bounding_box.as_lwh).astype(np.float32),
+        "heading": np.float32(val.heading),
+        "lane_index": np.int8(val.lane_index),
+        "linear_acceleration": val.linear_acceleration.astype(np.float32),
+        "linear_jerk": val.linear_jerk.astype(np.float32),
+        "linear_velocity": val.linear_velocity.astype(np.float32),
+        "pos": val.position.astype(np.float32),
```

Comment:
I think our positions are `np.float64` in the core.  Is that supported in spaces?  If so, we probably shouldn't lose precision here.

Reply:
Granted.

Reply:
Cool.  Now I'm wondering about some of the other `np.float32` fields as well.  Did you audit these to see if any of them should also be `np.float64`?

Reply:
I think the other fields are okay to use `np.float32`.